guides/source/action_mailer_basics.md

```diff
@@ -468,181 +447,220 @@ class UserMailer < ApplicationMailer
-special URL that renders them. In the above example, the preview class for
-`UserMailer` should be named `UserMailerPreview` and located in
-`test/mailers/previews/user_mailer_preview.rb`. To see the preview of
-`welcome_email`, implement a method that has the same name and call
-`UserMailer.welcome_email`:
+You either configure the `:host` option globally or make sure to pass it to the
+URL helper:
 
-```ruby
-class UserMailerPreview < ActionMailer::Preview
-  def welcome_email
-    UserMailer.with(user: User.first).welcome_email
-  end
-end
+```erb
+<%= user_url(@user, host: 'example.com') %>
 ```
 
-Then the preview will be available in <http://localhost:3000/rails/mailers/user_mailer/welcome_email>.
+### Adding Images in Action Mailer Views
 
-If you change something in `app/views/user_mailer/welcome_email.html.erb`
-or the mailer itself, it'll automatically reload and render it so you can
-visually see the new style instantly. A list of previews are also available
-in <http://localhost:3000/rails/mailers>.
+In order to use the `image_tag` helper in emails, you need to specify the
+`:asset_host` parameter. This is again because a mailer instance doesn't have
+any context about the incoming request.
 
-By default, these preview classes live in `test/mailers/previews`.
-This can be configured using the `preview_paths` option. For example, if you
-want to add `lib/mailer_previews` to it, you can configure it in
-`config/application.rb`:
+As the `:asset_host` usually is consistent across the application you can
+configure it globally in `config/application.rb`:
 
 ```ruby
-config.action_mailer.preview_paths << "#{Rails.root}/lib/mailer_previews"
+config.action_mailer.asset_host = 'http://example.com'
 ```
 
-### Generating URLs in Action Mailer Views
-
-Unlike controllers, the mailer instance doesn't have any context about the
-incoming request so you'll need to provide the `:host` parameter yourself.
+NOTE: Because we can't infer the protocol from the request, you'll need to
+specify a protocol such as `http://` or `https://` in the `:asset_host` config.
 
-As the `:host` usually is consistent across the application you can configure it
-globally in `config/application.rb`:
+Now you can display an image inside your email.
 
-```ruby
-config.action_mailer.default_url_options = { host: 'example.com' }
+```html+erb
+<%= image_tag 'image.jpg' %>
 ```
 
-Because of this behavior, you cannot use any of the `*_path` helpers inside of
-an email. Instead, you will need to use the associated `*_url` helper. For example
-instead of using
+### Caching Mailer View
+
+You can perform fragment caching in mailer views, similar to application views, using the [`cache`][] method.
 
 ```html+erb
-<%= link_to 'welcome', welcome_path %>
+<% cache do %>
+  <%= @company.name %>
+<% end %>
 ```
 
-You will need to use:
+And to use this feature, you need to enable it in your application's `config/application.rb` file:
```

Comment:
In a new Rails app this is now configured in the environment files:
https://github.com/rails/rails/blob/69c86f9606e490b24e8628dc47f34fbec6b321d1/railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt#L46
https://github.com/rails/rails/blob/69c86f9606e490b24e8628dc47f34fbec6b321d1/railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt#L91
```suggestion
And to use this feature, you need to enable it in your application's `config/environments/*.rb` file:
```